src/java.base/share/classes/jdk/internal/foreign/abi/Binding.java

```diff
@@ -387,6 +391,18 @@ public Binding.Builder dup() {
             return this;
         }
 
+        public Binding.Builder shiftLeft(int shiftAmount, Class<?> type) {
+            if (shiftAmount > 0 && isSubIntType(type)) {
+                bindings.add(Binding.cast(type, int.class));
+                type = int.class;
+            }
```

Comment:
I guess we would need to add additional bindings for that? Is is worth adding more just for a big endian corner case? Or can that be done with the existing ones?

Reply:
A `Cast` case for int -> long and long -> int would need to be added. Given the existing setup, that should only be a few lines of code for each. (See e.g. for int -> long https://github.com/openjdk/jdk/compare/master...JornVernee:jdk:I2L). I don't think the cost is that high.

> Is is worth adding more just for a big endian corner case?

I think it's worth it in order to have a cleaner contract for the shift ops, should we want to use them for anything else in the future, but also just to make them easier to understand for future readers.

Reply:
> I think it's worth it in order to have a cleaner contract for the shift ops, should we want to use them for anything else in the future, but also just to make them easier to understand for future readers.

I agree that having a cleaner contract for the shift binding would prove useful in the long run. If we do that, we can also simplify the binding itself, as it would no longer need an input type?

Reply:
> as it would no longer need an input type?

Yes. Then both shift ops would always operate on `long`.